### ISSUE TITLE
Windows - ghcjs-boot --dev

### ISSUE THREAD

**Issue by BethAr** [ghcjs/ghcjs#340]
Almost got there, but had another problem. Cabal failed.

 ```bash
Configuring Cabal-1.22.2.0...
Building Cabal-1.22.2.0...
Failed to install Cabal-1.22.2.0
JavaScript exception: Error: spawn EBADFcabal.exe: Error: some packages failed to install:
Cabal-1.22.2.0 failed during the building phase. The exception was:
ExitFailure 1
ghcjs-boot.exe: error running: C:\Users\Beta\AppData\Roaming\cabal\bin\cabal.exe --config-file C:\Users\Beta\AppData\Roaming\ghcjs\i386-mingw32-0.1.0-7.10.1\ghcjs\cabalBootConfig install ./boot/cabal/Cabal --global --ghcjs --one-shot --avoid-reinstalls --builddir dist --with-compiler C:\Users\Beta\AppData\Roaming\cabal\bin\ghcjs.exe --with-hc-pkg C:\Users\Beta\AppData\Roaming\cabal\bin\ghcjs-pkg.exe --prefix C:\Users\Beta\AppData\Roaming\ghcjs\i386-mingw32-0.1.0-7.10.1\ghcjs --enable-documentation --haddock-html --haddock-hyperlink-source --enable-library-profiling --disable-executable-stripping --disable-library-stripping
exit status: 1
stderr: Warning: No remote package servers have been specified. Usually you would have
one specified in the config file.
JavaScript exception: Error: spawn EBADFcabal.exe: Error: some packages failed to install:
Cabal-1.22.2.0 failed during the building phase. The exception was:
ExitFailure 1
```

**Comment by BethAr** [comment]
I uninstalled everything, removed all Haskell libraries, cleared my temp folders, downgraded GHC to 7.8.4, updated Cabal and cabal-install to 1.22, reinstalled Msys2 and everything it needs and recloned the GHCJS repo. I've set all enviroment variables and checked that they are ok.

GHCJS build completely correct, I used the flags:
`--reorder-goals --max-backjumps=-1 --constraint=transformers==0.3.0.0`
I checked that GHC had transformers 0.3.0.0 installed.

GHCJS-Boot --dev failed (as always):
```
cabal.exe: Error: some packages failed to install:
transformers-0.4.3.0 failed during the building phase. The exception was:
ExitFailure 1
```
How can this be?? I've set the constraint for transformers on the build! The culprit was probably Template Haskell as it was the last thing listed, but what was the point of the constraint in the first place?

After two days trying to build GHCJS, I give up.

**Comment by BethAr** [comment]
I tried one last time, did all the configuring steps above, but now I made sure transformers 0.3.0 was installed in GHC, and 0.4.3 was installed in Cabal.

I used the constraint and GHCJS built correctly. GHCJS-Boot, as usual, failed.

**Comment by asciiascetic** [comment]
Hail!  Your frustration is shared I'm sure, by me and surely others.

If it's any consolation, luite has given some indication (buried in a issue thread the link to which I cannot find just now) that the issue should resolve itself once ghc version bumps to 7.10.2.  This hopefully blessed state of affairs should occur [any day now](https://ghc.haskell.org/trac/ghc/blog/weekly20150414).  I'm content to wait it out.

**Comment by BethAr** [comment]
Thanks for the info @asciiascetic. I think I will be forced to step back now and wait.

I will leave the issue open as a warning for people.

**Comment by BethAr** [comment]
I managed to solve the problem in the following manner:
1. Installed the latest Cabal and cabal-install (which I tried before).
2. Use GHC 7.10.1 (which I also tried before).
3. Build GHCJS and GHCJS-Prim with the transformers constraint flag set to 0.4.2. (which I didn't even think of). I also used reorder-goals and max-backjumps as per the instructions.
4. Installed GHJCS-Boot from the 7.10 branch, as instructed.

On to the next problem.
